src/hooks/usePokemonDetails.ts

```diff
@@ -0,0 +1,16 @@
+import { useQuery } from "react-query";
+import { PokemonDetails } from "../interfaces/pokemonDetails";
+import { PokemonItem } from "../interfaces/pokemonItem";
+import { getPokemonDetails } from "../services/pokemonDetailsService";
+
+const usePokemonDetails = (pokemon: PokemonItem | null) => {
+  return useQuery<PokemonDetails | undefined, Error>(
```

Comment:
Perche' qui specifichi che puo' essere undefined?

Reply:
In effetti...